jdbc/src/main/java/org/apache/metamodel/jdbc/dialects/DB2QueryRewriter.java

```diff
@@ -69,11 +69,11 @@ public String rewriteQuery(Query query) {
         final Integer firstRow = query.getFirstRow();
         final Integer maxRows = query.getMaxRows();
 
-        if (maxRows == null && firstRow == null) {
+        if (maxRows == null && (firstRow == null || firstRow.intValue() == 1)) {
```

Comment:
Yes, but something went wrong - somehow @ankit2711's commit went into my PR ... Don't know why... at least it's not "my" change and wasn't my intention ... sorry, will repost I guess.